### DISCUSSION TITLE
Support for automatic filtering for consult commands where the search term is "#regexp#filter-string"

### DISCUSSION THREAD

**Discussion by geza-herman** [minad/consult#880] [Ideas]
Here's a problem for which I'd like to find a solution. I use [consult-lsp](https://github.com/gagbo/consult-lsp). It has a function: `consult-lsp-symbols`, which lists all the symbols of a project (using a language server). Some language servers, the search is fuzzy, and also case insensitive (for example, clangd seems to be such a server). So if I search for `Foo`, it will have `foo` in the results, which I don't want. My current solution is that I always narrow my search results down by repeating the search term: `#Foo#Foo`, as the second part of the search term is configured to be case sensitive.

I had the idea that consult can do this automatically in some way. Either it can be a setting (so `consult-lsp-symbols` can activate this mode), or consult could have a new specific separator. `#` could work as before, but there would be a new separator, like `!`, which would do the narrowing automatically. So, a search term of `!Foo` would be equivalent to `#Foo#Foo`. If `!Foo!<something>` is used, then `!` would work as `#` works (i.e., no special handling anymore).

So `!` would work like as `corfu`'s completion works with LSP: `corfu` automatically filters the result that the language server provides, so a completion of `Fo` doesn't have `foo` included as a possible candidate.

What do you think? Or is there some other kind of solution to this problem?

**Comment by minad** [comment]
I think it would be best if consult-lsp case sensitivity is configurable.

**Reply by geza-herman** [reply]
I think as long as the language server is not case sensitive, `consult-lsp` cannot be case sensitive either. As far as I understand, `consult-lsp` just passes the query to the language server as is. `consult-lsp` doesn't have a way to know how the server does the matching.
For example, if I enter the query `Foo`, the language server could return the symbol `full_object_orientation` (because the initials match). `consult-lsp` doesn't know that the `F` in `Foo` matches the first character of `full_object_orientation`, so it is not able to check whether the case matches (and in this case, as the case doesn't match so we should filter out this symbol).
